docs/DeveloperGuide.md

```diff
@@ -382,7 +429,42 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 1.  User requests to list persons
 2.  F.A.K.E.J.A.R.V.I.S. shows a list of persons
 3.  User requests to grade an assignment for a specific person in the list
-4.  F.A.K.E.J.A.R.V.I.S. assigns given grade to the student
+4.  F.A.K.E.J.A.R.V.I.S. assigns given grade to the student's assignment
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. F.A.K.E.J.A.R.V.I.S. shows an error message.
+
+      Use case resumes at step 2.
+
+* 3b. The given assignment name is invalid.
+
+    * 3b1. F.A.K.E.J.A.R.V.I.S. shows an error message.
+
+      Use case resumes at step 2.
+
+* 3c. The given grade is invalid.
+
+    * 3c1. F.A.K.E.J.A.R.V.I.S. shows an error message.
+
+      Use case resumes at step 2.
+
+**Use case: Delete the grade of an assignment**
+
+**MSS**
+
+1.  User requests to list persons
+2.  F.A.K.E.J.A.R.V.I.S. shows a list of persons
+3.  User requests to delete the grade of an assignment for a specific person in the list
+4.  F.A.K.E.J.A.R.V.I.S. deletes the grade of the student's assignment
```

Comment:
Any extensions for when grade is not assigned in the first place?

Reply:
Grade will be pre-assigned as ungraded